external-crates/move/move-compiler/src/cfgir/borrows/state.rs

```diff
@@ -164,25 +202,85 @@ impl BorrowState {
         Label::Resource(resource.value().to_owned())
     }
 
+    //**********************************************************************************************
+    // Metadata API
+    //**********************************************************************************************
+
+    pub fn start_command(&mut self, block: H::Label, command: usize) {
+        self.next_eid = ExpBasedID {
+            block: Some(block),
+            command,
+            count: 0,
+        };
+    }
+
+    fn set_param_name(&mut self, id: RefID, name: Var) {
+        let exps = self.id_to_exp.get(&id).unwrap();
+        assert_eq!(exps.len(), 1);
+        let infos: &mut BTreeMap<ExpBasedID, RefExpInfo> =
+            &mut RefCell::borrow_mut(&self.mutably_used);
+        for e in exps {
+            let info = infos.get_mut(e).unwrap();
+            assert!(info.param_name.is_none());
+            info.param_name = Some(name)
+        }
```

Comment:
Nit: Either there shouldn't be an assert here, or assert remains and the `for` loop should be rewritten to just directly access the zero-th exp so something like
```rust
let info = infos.get_mut(&exps[0]).unwrap();
...
```